autoload/ale/list.vim

```diff
@@ -0,0 +1,46 @@
+" Author: Bjorn Neergaard <bjorn@neersighted.com>, modified by Yann fery <yann@fery.me>
+" Description: Manages the loclist and quickfix lists
+
+function! ale#list#SetLists(loclist) abort
+    if g:ale_set_quickfix
+        call setqflist(a:loclist)
+    elseif g:ale_set_loclist
+        call setloclist(0, a:loclist)
+    endif
+
+    " If we don't auto-open lists, bail out here.
+    if !g:ale_open_list
+        return
+    endif
+
+    " If we have errors in our list, open the list.
+    if len(a:loclist) > 0
+        let l:winnr = winnr()
+        " There is some flicker with lopen and copen
+        " lopen Doc said 'If the window is already open and there are no recognized 
+        " errors, close the window.' but that doesn't seem to work that way
+        " lclose seems to be needed
+        if g:ale_set_quickfix
+            cwindow
+            " Only for vader testing
+            let g:ale_quickfix_opened = 1
+        elseif g:ale_set_loclist
+            lwindow
+            " Only for vader testing
+            let g:ale_loclist_opened = 1
```

Comment:
You're right. 

Before using those variables, I search a lot to find some way to test if loclist is opened or closed but I wasn't able to.
I'll keep looking a better way.